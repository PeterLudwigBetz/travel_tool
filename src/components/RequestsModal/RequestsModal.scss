.modal {
  &__user-info {
    margin: 10px;
  }
  &__modal1 {
    margin: 0 10px 20px 0px;
  }
  &__modal-date {
    display: flex;
    align-items: center;
    justify-content: flex-start;
  }
  &__travel-date {
    margin: 10px 55px 20px 5px;
  }
  &__travel-date:last-child {
    margin-right: 5px !important;
  }
  &__dialog {
    margin: 5px 0 10px 0;
  }
  &__dialog-btn {
    margin: 5px 20px 10px 0;
  }
  &__dialog2 {
    margin: 40px;
  }
  &__dialog3 {
    margin-right: 82px;
  }
  &__modal2 {
    margin-bottom: 30px;
    width: 100%;
    padding-left: 35px;
    border-bottom: none;
    box-sizing: border-box;
  }
  &__oval {
    height: 50px;
    width: 50px;
    margin: 15px 19px 0 0;
  }
  &__oval-copy {
    height: 30px;
    width: 30px;
    margin: 0 5px 0 0;
  }
  &__add-comment {
    height: 20px;
    width: 105px;
    color: #999999;
    font-family: 'DIN Pro';
    font-size: 15px;
    font-weight: 500;
    line-height: 20px;
    margin: 0 0 0 5px;
  }
  &__text-size {
    height: 20px;
    width: 85px;
    color: #4f4f4f;
    font-family: 'DIN Pro';
    font-size: 15px;
    font-weight: 500;
    line-height: 20px;
    margin: 0 0 0 0;
  }
  &__modal3 {
    height: 18px;
    color: #999999;
    font-family: 'DIN Pro';
    font-size: 14px;
    line-height: 18px;
    margin: -20px 0 0 70px;
  }
  &__dialog1 {
    height: 20px;
    width: 194px;
    color: #4f4f4f;
    font-family: 'DIN Pro';
    font-size: 15px;
    font-weight: 500;
    line-height: 20px;
    margin: 5px 15px 10px 0;
  }
  &__dialog-from {
    height: 20px;
    width: 89px;
    color: #4f4f4f;
    font-family: 'DIN Pro';
    font-size: 15px;
    font-weight: 500;
    line-height: 20px;
    margin: 5px 50px 10px 0;
  }
  &__travel-place {
    margin: 40px 5px 5px 5px;
  }
  &__travel-dates {
    color: #999999;
    font-family: 'DIN Pro';
    font-size: 14px;
    line-height: 18px;
  }
  &__departure-date {
    height: 18px;
    width: 137px;
    color: #999999;
    font-family: 'DIN Pro';
    font-size: 14px;
    line-height: 18px;
    margin: 5px 50px 10px 0;
  }
  &__city-name {
    color: #3359db;
    padding-left: 5px;
    font-weight: 1000;
  }

  &__date-text {
    color: #4f4f4f;
    font-family: 'DIN Pro';
    font-size: 15px;
    font-weight: 500;
    line-height: 20px;
    padding-top: 7px;
  }

  &__button-submitted-text {
    box-sizing: border-box;
    height: 29px;
    width: 94px;
    border: 1px solid #d4d4d4;
    border-radius: 1px;
    background-color: #ffffff;
    color: #4f4f4f;
    font-family: 'DIN Pro';
    font-size: 14px;
    font-weight: 500;
    line-height: 18px;
    text-align: center;
    border-radius: 1px;
  }

  &__button-rejected-text {
    box-sizing: border-box;
    height: 29px;
    width: 81px;
    border: 1px solid #d4d4d4;
    border-radius: 1px;
    background-color: #ffffff;
    color: #4f4f4f;
    font-family: 'DIN Pro';
    font-size: 14px;
    font-weight: 500;
    line-height: 18px;
    text-align: center;
    border-radius: 1px;
  }
  
  button:focus {
    outline: none;
  }
  &__button-below {
    margin: 19.5px 0 60px 5px;
  }

  &__target-return-date {
    height: 18px;
    width: 115px;
    color: #999999;
    font-family: 'DIN Pro';
    font-size: 14px;
    line-height: 18px;
    margin: 5px 50px 10px 0;
  }

  &__approval-status {
    color: rgb(102, 101, 101);
    font-family: 'DIN Pro';
    font-size: 15px;
    line-height: 20px;
    margin-right: 5px;
  }
  &__hours-status {
    color:#bcbcbc;
    font-family: 'DIN Pro';
    font-size: 15px;
    line-height: 20px;
    text-align: right;
  }
  &__user-name {
    color: #3359db;
    font-family: 'DIN Pro';
    font-size: 15px;
    line-height: 20px;
    margin-right: 5px;
  }
  &__delete-btn {
    height: 15px;
    width: 35px;
    color: #999999;
    font-family: 'DIN Pro';
    font-size: 12px;
    line-height: 15px;
    text-align: right;
    padding: 0;
    border: none;
    background: none;
    margin-left: 20px;
  }
  &__status-update {
    color: rgb(102, 101, 101);
    font-family: 'DIN Pro';
    font-size: 15px;
    line-height: 20px;
  }
  hr {
    position: absolute;
    margin: -9px -9px -9px 0;
    width: 90%;
    height: 3px;
    border: 1px solid #f8f8f8;
    border-bottom: none;
    box-sizing: border-box;
  }
}
