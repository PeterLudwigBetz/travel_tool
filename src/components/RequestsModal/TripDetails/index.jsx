import React, {PureComponent, Fragment} from 'react';
import {PropTypes} from 'prop-types';
import generateDynamicDate from '../../../helper/generateDynamicDate';
import TripDetail from './TripDetail';

export default class TripDetails extends PureComponent {
  static propTypes = {
    tripDetails: PropTypes.object.isRequired
  }
  renderTripDetails = (
    origin, destination, tripDetails, departureDate, tripType, returnDate, accomodationdetails
  ) => (
    <React.Fragment>
      <div>
        <TripDetail
          label="From"
          value={origin} />
      </div>
      <div>
        <TripDetail
          label="Travel to"
          value={destination} />
      </div>
      <div>
        <TripDetail
          label="Depature date"
          value={generateDynamicDate(tripDetails, departureDate)} />
      </div>
      <div id="return-date">
        { tripType !== 'oneWay' ?
          (<TripDetail
            label="Return date"
            value={generateDynamicDate(tripDetails, returnDate)} />
          )  : ''
        }
      </div>
      <div id="trip-detail">
        <TripDetail
          label="Accomodation"
          value={accomodationdetails} />
      </div>
    </React.Fragment>
  )
  renderTrips() {
    const { tripDetails } = this.props;
    const {
      beds,
      createdAt,
      departureDate,
      returnDate,
      destination,
      origin,
      tripType
    } = tripDetails;
    const {rooms:{roomName, guestHouses:{houseName}}, bedName } = beds;
    const accomodationdetails = `${houseName}, ${roomName}, ${bedName}`;
   
    return (
      <div className={`modal__modal-trip-details ${tripType}`}>
        {this.renderTripDetails(
          origin, destination, tripDetails, departureDate, tripType, returnDate, accomodationdetails
        )}
      </div>
    );
  }

  render() {
    return this.renderTrips();
  }
}
