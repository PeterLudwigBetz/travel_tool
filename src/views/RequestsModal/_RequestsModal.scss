.mce-notification.mce-in {
  display: none;
}

#mceu7 {
  height: 20px;
  width: 115px;
  color: #999999;
  font-family: 'DIN Pro';
  font-size: 15px;
  line-height: 20px;
  text-align: right;
}

#mceu_2-button {
  margin-right: 8px;
  color: #999999;
}

#mceu_4 {
  margin-right: 8px;
}
#mceu_7 {
  border-color: #e4e4e4;
  border-width: 0;
  border-radius: 2px;
  box-shadow: inset 0 -1px 0 0 #f8f8f8;
}
#mceu_12 {
  border-color: #e4e4e4;
  border-width: 50px;
}
.mce-panel {
  border: 0;
  border-width: 0;
}
.mce-ico::before {
  height: 16px;
  width: 16px;
  color: #999999;
}

.modal {
  &__modal_dialog {
    margin: 10px;
  }
  &__modal1 {
    margin: 10px 10px 20px 10px;
  }

  &__dialog {
    margin: 5px 50px 10px 0;
  }
  &__dialog-btn {
    margin: 5px 20px 10px 0;
  }
  &__dialog2 {
    margin: 40px;
  }
  &__dialog3 {
    margin-right: 82px;
  }
  &__modal2 {
    margin-bottom: 30px;
    width: 100%;
    padding-left: 47px;
    border-bottom: none;
    box-sizing: border-box;
  }
  &__oval {
    height: 50px;
    width: 50px;
    margin: 15px 19px 0 0;
  }
  &__oval-copy {
    height: 30px;
    width: 30px;
    margin: 0 5px 0 -10px;
  }
  &__add-comment {
    height: 20px;
    width: 105px;
    color: #999999;
    font-family: 'DIN Pro';
    font-size: 15px;
    font-weight: 500;
    line-height: 20px;
    margin: 0 0 0 5px;
  }
  &__text-size {
    height: 20px;
    width: 85px;
    color: #4f4f4f;
    font-family: 'DIN Pro';
    font-size: 15px;
    font-weight: 500;
    line-height: 20px;
    margin: 0 0 0 0;
  }
  &__modal3 {
    height: 18px;
    color: #999999;
    font-family: 'DIN Pro';
    font-size: 14px;
    line-height: 18px;
    margin: -15px 0 0 70px;
  }
  &__dialog1 {
    height: 20px;
    width: 194px;
    color: #4f4f4f;
    font-family: 'DIN Pro';
    font-size: 15px;
    font-weight: 500;
    line-height: 20px;
    margin: 5px 50px 10px 0;
  }
  &__dialog-from {
    height: 20px;
    width: 89px;
    color: #4f4f4f;
    font-family: 'DIN Pro';
    font-size: 15px;
    font-weight: 500;
    line-height: 20px;
    margin: 5px 50px 10px 0;
  }
  &__travel-place {
    margin: 40px 5px 20px 5px;
  }
  &__date-submitted {
    height: 18px;
    width: 95px;
    color: #999999;
    font-family: 'DIN Pro';
    font-size: 14px;
    line-height: 18px;
    margin: 5px 50px 10px 0;
  }
  &__departure-date {
    height: 18px;
    width: 137px;
    color: #999999;
    font-family: 'DIN Pro';
    font-size: 14px;
    line-height: 18px;
    margin: 5px 50px 10px 0;
  }
  &__date-date {
    height: 20px;
    width: 82px;
    color: #4f4f4f;
    font-family: 'DIN Pro';
    font-size: 15px;
    font-weight: 500;
    line-height: 20px;
    margin-right: 62px;
  }
  &__date-date2 {
    height: 20px;
    width: 86px;
    color: #4f4f4f;
    font-family: 'DIN Pro';
    font-size: 15px;
    font-weight: 500;
    line-height: 20px;
    margin-left: 45px;
  }

  &__button-submitted-text {
    box-sizing: border-box;
    height: 29px;
    width: 94px;
    border: 1px solid #d4d4d4;
    border-radius: 1px;
    background-color: #ffffff;
    color: #4f4f4f;
    font-family: 'DIN Pro';
    font-size: 14px;
    font-weight: 500;
    line-height: 18px;
    text-align: center;
  }

  &__button-rejected-text {
    box-sizing: border-box;
    height: 29px;
    width: 81px;
    border: 1px solid #d4d4d4;
    border-radius: 1px;
    background-color: #ffffff;
    color: #4f4f4f;
    font-family: 'DIN Pro';
    font-size: 14px;
    font-weight: 500;
    line-height: 18px;
    text-align: center;
  }
  button:focus {
    outline: none;
  }
  &__button-below {
    margin: 10px 10px 70px 10px;
  }

  &__target-return-date {
    height: 18px;
    width: 115px;
    color: #999999;
    font-family: 'DIN Pro';
    font-size: 14px;
    line-height: 18px;
    margin: 5px 50px 10px 0;
  }

  &__editor {
    height: 125px;
    width: 80%;
    margin: 0 0 90px 0;
    // border-radius: 2px;
    // background-color: #FFFFFF;
  }

  &__post-btn {
    box-sizing: border-box;
    height: 29px;
    width: 70px;
    border: 1px solid #d4d4d4;
    border-radius: 1px;
    background-color: #ffffff;
    color: #999999;
    font-family: 'DIN Pro';
    font-size: 14px;
    font-weight: 500;
    line-height: 18px;
    text-align: center;
    margin: 20px 0 20px -8px;
  }
  &__approval-status {
    height: 20px;
    width: 307px;
    color: #4f4f4f;
    font-family: 'DIN Pro';
    font-size: 15px;
    line-height: 20px;
    margin-right: 5px;
  }
  &__hours-status {
    height: 20px;
    width: 78px;
    color: #bcbcbc;
    font-family: 'DIN Pro';
    font-size: 15px;
    line-height: 20px;
    text-align: right;
  }
  &__user-name {
    height: 20px;
    width: 107px;
    color: #3359db;
    font-family: 'DIN Pro';
    font-size: 15px;
    line-height: 20px;
    margin-right: 5px;
  }
  &__delete-btn {
    height: 15px;
    width: 35px;
    color: #999999;
    font-family: 'DIN Pro';
    font-size: 12px;
    line-height: 15px;
    text-align: right;
    padding: 0;
    border: none;
    background: none;
    margin-left: 20px;
  }
  &__status-update {
    height: 20px;
    width: 384px;
    color: #4f4f4f;
    font-family: 'DIN Pro';
    font-size: 15px;
    line-height: 20px;
    margin: -10px 5px 20px -10px;
  }
  hr {
    position: absolute;
    margin: -9px -9px -9px 0;
    width: 90%;
    height: 3px;
    border: 1px solid #f8f8f8;
    border-bottom: none;
    box-sizing: border-box;
  }
  
}
