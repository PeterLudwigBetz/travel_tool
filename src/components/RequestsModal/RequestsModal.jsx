import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Oval2 from '../../images/Oval2.png';
import ConnectedCommentBox from './CommentBox/CommentBox';
import ImageLink from '../image-link/ImageLink';
import { fetchUserRequestDetails } from '../../redux/actionCreator/requestActions';
import { updateRequestStatus } from '../../redux/actionCreator/approvalActions';
import './RequestsModal.scss';
import AddComment from './CommentBox/AddComment';
import ConnectedUserComments from './UserComments/UserComments';
import UserInfo from './UserInfo/UserInfo';
import TripDetails from './TripDetails';
import RequestDetailsHeader from './RequestDetailsHeader';

export class RequestDetailsModal extends Component {
  state = {
    approveColor: '',
    rejectColor: '',
    approveTextColor: '',
    rejectTextColor: '',
    buttonSelected: false,
    approvalText: 'Approve',
    rejectText: 'Reject'
  };

  componentDidMount() {
    const { fetchUserRequestDetails, requestId } = this.props;
    fetchUserRequestDetails(requestId);
  }

  getRequestTripsDetails = (requestData) => {
    const {trips, tripType, createdAt} = requestData;
    const requestTripsDetails = trips && trips.map(trip => {
      const tripDetails = { createdAt, tripType, ...trip, };
      return <TripDetails key={trip.id} tripDetails={tripDetails} />;
    });
    return requestTripsDetails;
  }

  handleButtonState = (approvalText, rejectText, buttonSelected, approveColor, rejectColor, approveTextColor, rejectTextColor, newStatus, requestId) => {
    this.setState({
      approvalText,
      rejectText,
      approveColor,
      rejectColor,
      approveTextColor,
      rejectTextColor,
      buttonSelected,
      newStatus,
      requestId }, () => this.updateRequestStatus());
  }

  changeButtonColor = (button, status) => {
    const { approveTextColor, rejectTextColor } = this.state;
    const style = {
      color: 'white'
    };
    const approvedCondition = approveTextColor && button.id === 1 && !rejectTextColor;
    const rejectCondition = rejectTextColor && button.id === 2 && !approveTextColor;

    if ((status === 'Approved' && button.id === 1) || approvedCondition) {
      style.backgroundColor = '#49AAAF';
      return style;
    } else if ((status === 'Rejected' && button.id === 2) || rejectCondition) {
      style.backgroundColor = '#FF5359';
      return style;
    } else {
      style.color = 'black';
      style.backgroundColor = 'white';
      return style;
    }
  }

  handleApprove = (requestId) => {
    return () => {
      this.handleButtonState('Approved','Reject',true,'#49AAAF', '', 'white', '','Approved', requestId);
    };
  };

  handleReject = (requestId) => {
    return () => {
      this.handleButtonState('Approve', 'Rejected', true, '', '#FF5359', '', 'white', 'Rejected', requestId);
    };
  };

  showButtons = (approveColor, rejectColor, approveTextColor, rejectTextColor, id) => {
    const { approvalText, rejectText } =  this.state;
    return (
      [
        {
          id: 1,
          onClick: this.handleApprove(id),
          action: approveColor,
          actionText: approveTextColor,
          class: 'modal__button-submitted-text bg',
          text: approvalText
        },
        {
          id: 2,
          onClick: this.handleReject(id),
          action: rejectColor,
          actionText: rejectTextColor,
          class: 'modal__button-rejected-text',
          text: rejectText
        }
      ]);
  }

  updateRequestStatus = () => {
    const { requestId, newStatus }= this.state;
    const { updateRequestStatus } = this.props;
    updateRequestStatus({requestId, newStatus});
  }


  disableButtons(status, page) {
    const { buttonSelected } = this.state;
    const { isStatusUpdating } = this.props;
    return page === 'Requests' ||
      page === 'Approvals' && (status && ['Approved', 'Rejected'].includes(status)) ||
      isStatusUpdating || buttonSelected;
  }
  shouldButtonsRender(status) {
    const { navigatedPage } = this.props;
    if (navigatedPage === 'Requests' && (status === 'Open' ||'Approved' || 'Rejected')) return this.renderStatusAsBadge(status);
    else return this.renderButtons();
  }

  renderButtonText(status, text) {
    return status && status.includes(text) ? status : text;
  }

  renderStatusAsBadge(status) {
    const style = `request__status--${!status ? '' : status.toLowerCase()}`;
    return (
      <div className="modal__button-below">
        <span className={style}>
          {status}
        </span>
      </div>
    );
  }

  renderRequestDetailsHeader = () => {
    const {requestData} = this.props;
    return <RequestDetailsHeader requestData={requestData} />;
  }

  renderButtons = () => {
    const { approveColor, rejectColor, approveTextColor, rejectTextColor } = this.state;
    const { requestData: { id, status }, navigatedPage } = this.props;
    let displayButtons = this.showButtons(approveColor, rejectColor, approveTextColor, rejectTextColor, id)
      .map((button)=>{
        const buttonStyle = this.changeButtonColor(button, status);
        return (
          <span key={button.text}>
            <span className="modal__dialog-btn">
              <button
                style={{
                  backgroundColor: `${buttonStyle.backgroundColor}`,
                  color: `${buttonStyle.color}`, cursor: status === 'Open' ? 'Pointer' : ''
                }}
                onClick={button.onClick}
                className={button.class}
                id={`b${button.id}`}
                type="button"
                disabled={this.disableButtons(status, navigatedPage)}
              >
                {this.renderButtonText(status, button.text)}
              </button>
            </span>
          </span>
        );
      });

    return (
      <div className="modal__button-below">
        {displayButtons}
      </div>
    );
  }

  renderRequestAprroval = () => {
    const { requestData } = this.props;
    return (
      <div className="modal__modal1">
        <span className="modal__mdl-icons">
          <ImageLink
            imageSrc={Oval2}
            altText="avatar"
            imageClass="modal__oval-copy" />
          <span className="modal__user-name">
            {requestData && requestData.manager}
          </span>
          <span className="modal__approval-status">
          approved your travel request.
          </span>
          <span className="modal__hours-status">
           5 hours ago
          </span>
        </span>
      </div>
    );
  }

  render() {
    const { requestId, requestData, user, email } = this.props;
    const { status, comments } = requestData;
    const { picture } = user;
    return (
      <Fragment>
        <div style={{display:'flex', flexWrap:'wrap', justifyContent: 'space-between'}}>
          <UserInfo
            requestData={requestData}
            user={user}
          />
          {this.shouldButtonsRender(status)}
        </div>
        <div className="request-details">
          {this.renderRequestDetailsHeader(requestData)}
          {this.getRequestTripsDetails(requestData)}
        </div>
        <AddComment
          image={picture}
        />
        <ConnectedCommentBox requestId={requestId} />
        {(requestData && status) === 'Approved' && this.renderRequestAprroval()}
        <div id="comments">
          <ConnectedUserComments comments={comments} email={email} />
        </div>
      </Fragment>
    );
  }
}

RequestDetailsModal.propTypes = {
  fetchUserRequestDetails: PropTypes.func,
  updateRequestStatus: PropTypes.func,
  requestId: PropTypes.string,
  user: PropTypes.object,
  requestData: PropTypes.object,
  isStatusUpdating: PropTypes.bool,
  navigatedPage: PropTypes.string,
  email:PropTypes.string
};

RequestDetailsModal.defaultProps = {
  fetchUserRequestDetails: () => {},
  updateRequestStatus: () => {},
  requestId: '',
  requestData: {},
  user: {},
  isStatusUpdating: false,
  navigatedPage: '',
  email: ''
};

const mapStateToProps = (state) => {
  return {
    requestData: state.requests.requestData,
    user: state.auth.user.UserInfo,
    isStatusUpdating: state.approvals.updatingStatus,
    email:state.user.getUserData.result.email,
    ...state.modal.modal
  };
};

const actionCreators = {
  fetchUserRequestDetails,
  updateRequestStatus
};

export default connect(mapStateToProps, actionCreators)(RequestDetailsModal);
