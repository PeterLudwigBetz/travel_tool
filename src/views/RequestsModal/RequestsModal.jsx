import React, { Component } from 'react';
import { Editor } from '@tinymce/tinymce-react';
import Oval from '../../images/Oval.png';
import Oval2 from '../../images/Oval2.png';
import './_RequestsModal.scss';

class RequestDetailsModal extends Component {
  state = {
    approveColor: '',
    rejectColor: '',
    approveTextColor: '',
    rejectTextColor: '',
    dataInput: ''
  };

  handleApprove = () => {
    this.setState({
      approveColor: '#49AAAF',
      rejectColor: '',
      approveTextColor: 'white',
      rejectTextColor: ''
    });
  };

  handleReject = () => {
    this.setState({
      approveColor: '',
      rejectColor: '#FF5359',
      approveTextColor: '',
      rejectTextColor: 'white'
    });
  };

  handleEditorChange = e => {
    e.preventDefault();
    this.setState({
      dataInput: console.log('Content was updated:', e.target.getContent())
    });
  };

  handleSubmit = e => {
    e.preventDefault();
    this.setState({
      dataInput: event.target.value
    });
    console.log(this.dataInput);
  };

  renderUserInfo() {
    return (
      <div>
        <div className="modal__modal_dialog">
          <img src={Oval} alt="avatar" className="modal__oval" />
          <span className="modal__text-size">
Silm Momoh
          </span>
          <div className="modal__modal3">
Product Designer, Apprenticeship
          </div>
        </div>
      </div>
    );
  }

  renderRequestInfo() {
    return (
      <div>
        <div className="modal__travel-place">
          <span className="modal__dialog1">
Request to travel to: Nairobi
          </span>
          <span className="modal__dialog-from">
From: Lagos
          </span>
        </div>
        <div className="modal__modal1">
          <span className="modal__date-submitted">
Date submitted
          </span>
          <span className="modal__departure-date">
Target departure date
          </span>
          <span className="modal__target-return-date">
Target return date
          </span>
        </div>
      </div>
    );
  }

  renderRequestDate() {
    return (
      <div>
        <div className="modal__modal1">
          <span className="modal__date-date">
02 Aug 2018
          </span>
          <span className="modal__date-date">
11 Aug 2018
          </span>
          <span className="modal__date-date2">
05 Sept 2018
          </span>
        </div>
      </div>
    );
  }

  renderButtons() {
    const {
      approveColor,
      rejectColor,
      approveTextColor,
      rejectTextColor
    } = this.state;
    return (
      <div>
        <div className="modal__button-below">
          <span className="modal__dialog-btn">
            <button
              style={{
                backgroundColor: `${approveColor}`,
                color: `${approveTextColor}`
              }}
              onClick={this.handleApprove}
              type="button"
              className="modal__button-submitted-text bg"
            >
              Approve
            </button>
          </span>
          <span className="modal__dialog-btn">
            <button
              style={{
                backgroundColor: `${rejectColor}`,
                color: `${rejectTextColor}`
              }}
              onClick={this.handleReject}
              type="button"
              className="modal__button-rejected-text bgg"
            >
              Reject
            </button>
          </span>
        </div>
        <div className="modal__modal1">
          <span className="modal__mdl-icons">
            <img src={Oval2} alt="avatar" className="modal__oval-copy" />
            <span className="modal__add-comment">
Add a comment
            </span>
          </span>
        </div>
      </div>
    );
  }

  renderCommentBox() {
    const { dataInput } = this.state;
    return (
      <form onSubmit={this.handleSubmit} className="modal__editor">
        <Editor
          id="my-id"
          initialValue="<p style='color:#999999; font-size: 15px;	font-family: DIN Pro;	line-height: 20px; text-align: left; margin:20px 8px;'>Write a comment</p>"
          init={{
            body_class:
              'mce-notification.mce-in mce-tinymce mce-container mce-panel mce-ico',
            body_id: '#mceu_0 #mceu_1 #mceu_2',
            statusbar: false,
            selector: 'textarea',
            plugins: 'lists',
            skin: 'lightgray',
            menubar: false,
            branding: false,
            toolbar: 'bold italic underline   numlist bullist   outdent indent'
          }}
          onChange={this.handleEditorChange}
          value={this.dataInput}
        />
        <div className="modal__modal1">
          <span className="modal__dialog">
            <button type="submit" className="modal__post-btn post-btn-text">
              Post
            </button>
          </span>
        </div>
      </form>
    );
  }

  renderRequestAprroval() {
    return (
      <div className="modal__modal1">
        <span className="modal__mdl-icons">
          <img src={Oval2} alt="avatar" className="modal__oval-copy" />
          <span className="modal__user-name">
Jolomi Otumara
          </span>
          <span className="modal__approval-status">
approved your travel request.
          </span>
          <span className="modal__hours-status">
5 hours ago
          </span>
        </span>
      </div>
    );
  }

  renderUserComments() {
    return (
      <div>
        <hr />
        <div className="modal__modal1">
          <span className="modal__mdl-icons">
            <img src={Oval2} alt="avatar" className="modal__oval-copy" />
            <span className="modal__user-name">
Jolomi Otumara.
            </span>
            <span className="modal__hours-status">
5 hours ago
            </span>
            <span className="modal__dialog">
              <button type="button" className="modal__delete-btn">
                Delete
              </button>
            </span>
          </span>
        </div>
        <div className="modal__modal2">
          <div className="modal__status-update">
            I thought we agreed you would be travelling Next Month?
          </div>
        </div>
      </div>
    );
  }
  render() {
    return (
      <div>
        {this.renderUserInfo()}
        {this.renderRequestInfo()}
        {this.renderRequestDate()}
        {this.renderButtons()}
        {this.renderCommentBox()}
        {this.renderRequestAprroval()}
        {this.renderUserComments()}
      </div>
    );
  }
}

export default RequestDetailsModal;
