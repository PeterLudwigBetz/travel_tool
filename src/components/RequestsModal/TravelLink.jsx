import React, { PureComponent } from 'react';

import PropTypes from 'prop-types';
import './RequestsModal.scss';

class TravelLink extends PureComponent {
  render() {
    const { divClass, innerClass, dynamicText,  nextClass, dynamicDate } = this.props;
    return (
      <div className={divClass}>
        <div className={innerClass}>
          {dynamicText}
        </div>
        <div className={nextClass}>
          {dynamicDate}
        </div>
      </div>
    );
  }
}

TravelLink.propTypes = {
  divClass: PropTypes.string.isRequired,
  innerClass: PropTypes.string.isRequired,
  dynamicText: PropTypes.string.isRequired,
  nextClass: PropTypes.string.isRequired,
  dynamicDate:PropTypes.string.isRequired
};

export default TravelLink;
