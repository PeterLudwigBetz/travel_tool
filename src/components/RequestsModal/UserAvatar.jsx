import React, { PureComponent } from 'react';

import PropTypes from 'prop-types';
import './RequestsModal.scss';
import '../nav-bar/NavBar.scss';

class UserAvatar extends PureComponent {
  render() {
    const { imageSrc, altText, imageClass, spanClass, spanTextClass, dynamicText } = this.props;
    return (
      <span className={spanClass}>
        <img src={imageSrc} alt={altText} className={imageClass} />
        <span className={spanTextClass}>
          {dynamicText}
        </span>
      </span>
    );
  }
}

UserAvatar.propTypes = {
  imageSrc: PropTypes.string.isRequired,
  altText: PropTypes.string.isRequired,
  imageClass: PropTypes.string.isRequired,
  spanClass: PropTypes.string.isRequired,
  spanTextClass:PropTypes.string.isRequired,
  dynamicText:PropTypes.string.isRequired
};

export default UserAvatar;
