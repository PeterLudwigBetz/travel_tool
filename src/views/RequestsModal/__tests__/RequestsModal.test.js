import React from 'react';
import { shallow } from 'enzyme';
import RequestsModal from '../RequestsModal';

// describe what we are testing
describe('Render RequestsModal component', () => {
  // const props = {
  //   onNotificationToggle: jest.fn(),
  //   avatar: 'avatar'
  // };
  // make our assertions and what we expect to happen
  it('should match snapshot', () => {
    const wrapper = shallow(<RequestsModal />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the RequestsModal as expected', () => {
    const wrapper = shallow(<RequestsModal />);
    expect(wrapper.length).toBe(1);
  });

  // it('should render the OnNotification click as exepected', () => {
  //   const wrapper = shallow(<NavBar {...props} />);
  //   let NotificationToggleSpy = jest.spyOn(
  //     wrapper.instance().props,
  //     'onNotificationToggle'
  //   );
  //   wrapper
  //     .find('#notification')
  //     .first()
  //     .simulate('click');
  //   expect(NotificationToggleSpy).toHaveBeenCalled();
  // });
});
