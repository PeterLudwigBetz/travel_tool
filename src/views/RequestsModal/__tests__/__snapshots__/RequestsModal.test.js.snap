// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render RequestsModal component should match snapshot 1`] = `
<div>
  <div>
    <div
      className="modal__modal_dialog"
    >
      <img
        alt="avatar"
        className="modal__oval"
        src="Oval.png"
      />
      <span
        className="modal__text-size"
      >
        Silm Momoh
      </span>
      <div
        className="modal__modal3"
      >
        Product Designer, Apprenticeship
      </div>
    </div>
  </div>
  <div>
    <div
      className="modal__travel-place"
    >
      <span
        className="modal__dialog1"
      >
        Request to travel to: Nairobi
      </span>
      <span
        className="modal__dialog-from"
      >
        From: Lagos
      </span>
    </div>
    <div
      className="modal__modal1"
    >
      <span
        className="modal__date-submitted"
      >
        Date submitted
      </span>
      <span
        className="modal__departure-date"
      >
        Target departure date
      </span>
      <span
        className="modal__target-return-date"
      >
        Target return date
      </span>
    </div>
  </div>
  <div>
    <div
      className="modal__modal1"
    >
      <span
        className="modal__date-date"
      >
        02 Aug 2018
      </span>
      <span
        className="modal__date-date"
      >
        11 Aug 2018
      </span>
      <span
        className="modal__date-date2"
      >
        05 Sept 2018
      </span>
    </div>
  </div>
  <div>
    <div
      className="modal__button-below"
    >
      <span
        className="modal__dialog-btn"
      >
        <button
          className="modal__button-submitted-text bg"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
              "color": "",
            }
          }
          type="button"
        >
          Approve
        </button>
      </span>
      <span
        className="modal__dialog-btn"
      >
        <button
          className="modal__button-rejected-text bgg"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
              "color": "",
            }
          }
          type="button"
        >
          Reject
        </button>
      </span>
    </div>
    <div
      className="modal__modal1"
    >
      <span
        className="modal__mdl-icons"
      >
        <img
          alt="avatar"
          className="modal__oval-copy"
          src="Oval2.png"
        />
        <span
          className="modal__add-comment"
        >
          Add a comment
        </span>
      </span>
    </div>
  </div>
  <form
    className="modal__editor"
    onSubmit={[Function]}
  >
    <Editor
      id="my-id"
      init={
        Object {
          "body_class": "mce-notification.mce-in mce-tinymce mce-container mce-panel mce-ico",
          "body_id": "#mceu_0 #mceu_1 #mceu_2",
          "branding": false,
          "menubar": false,
          "plugins": "lists",
          "selector": "textarea",
          "skin": "lightgray",
          "statusbar": false,
          "toolbar": "bold italic underline   numlist bullist   outdent indent",
        }
      }
      initialValue="<p style='color:#999999; font-size: 15px;	font-family: DIN Pro;	line-height: 20px; text-align: left; margin:20px 8px;'>Write a comment</p>"
      onChange={[Function]}
    />
    <div
      className="modal__modal1"
    >
      <span
        className="modal__dialog"
      >
        <button
          className="modal__post-btn post-btn-text"
          type="submit"
        >
          Post
        </button>
      </span>
    </div>
  </form>
  <div
    className="modal__modal1"
  >
    <span
      className="modal__mdl-icons"
    >
      <img
        alt="avatar"
        className="modal__oval-copy"
        src="Oval2.png"
      />
      <span
        className="modal__user-name"
      >
        Jolomi Otumara
      </span>
      <span
        className="modal__approval-status"
      >
        approved your travel request.
      </span>
      <span
        className="modal__hours-status"
      >
        5 hours ago
      </span>
    </span>
  </div>
  <div>
    <hr />
    <div
      className="modal__modal1"
    >
      <span
        className="modal__mdl-icons"
      >
        <img
          alt="avatar"
          className="modal__oval-copy"
          src="Oval2.png"
        />
        <span
          className="modal__user-name"
        >
          Jolomi Otumara.
        </span>
        <span
          className="modal__hours-status"
        >
          5 hours ago
        </span>
        <span
          className="modal__dialog"
        >
          <button
            className="modal__delete-btn"
            type="button"
          >
            Delete
          </button>
        </span>
      </span>
    </div>
    <div
      className="modal__modal2"
    >
      <div
        className="modal__status-update"
      >
        I thought we agreed you would be travelling Next Month?
      </div>
    </div>
  </div>
</div>
`;
